tasks/publish-pyxis-repository/tests/test-publish-pyxis-repository-source-container-mapping.yaml

```diff
@@ -0,0 +1,106 @@
+          workspace: tests-workspace
+      taskSpec:
+        workspaces:
+          - name: data
+        steps:
+          - name: setup-values
+            image: quay.io/redhat-appstudio/release-service-utils:bc81bfed6062a386e48a76b252c6f33b52c411b0
+            script: |
+              #!/usr/bin/env sh
+              set -eux
+
+              cat > $(workspaces.data.path)/snapshot_spec.json << EOF
+              {
+                "application": "my-app",
+                "components": [
+                  {
+                    "repository": "quay.io/redhat-prod/my-product----my-image1",
+                    "pushSourceContainer": "true"
+                  },
+                  {
+                    "repository": "quay.io/redhat-prod/my-product----my-image2",
+                    "pushSourceContainer": "false"
+                  }, 
+                  {
+                    "repository": "quay.io/redhat-prod/my-product----my-image3"
+                  }
+                ]
+              }
+              EOF
+
+              cat > $(workspaces.data.path)/mydata.json << EOF
+              {
+                "mapping": {
+                  "defaults": {
+                    "pushSourceContainer": "true"
+                  }
+                }
+              }
+              EOF
+    - name: run-task
+      taskRef:
+        name: publish-pyxis-repository
+      params:
+        - name: pyxisSecret
+          value: test-publish-pyxis-repository-cert
+        - name: snapshotPath
+          value: snapshot_spec.json
+        - name: dataPath
+          value: mydata.json
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+      runAfter:
+        - setup
+    - name: check-result
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+      taskSpec:
+        workspaces:
+          - name: data
+        steps:
+          - name: check-result
+            image: quay.io/redhat-appstudio/release-service-utils:bc81bfed6062a386e48a76b252c6f33b52c411b0
+            script: |
+              #!/usr/bin/env sh
+              set -eux
+
+              if [ $(cat $(workspaces.data.path)/mock_curl.txt | wc -l) != 6 ]; then
+                  echo Error: curl was expected to be called 6 times. Actual calls:
+                  cat $(workspaces.data.path)/mock_curl.txt
+                  exit 1
+              fi
+
+              [[ $(cat $(workspaces.data.path)/mock_curl.txt | head -n 1) \
+                  == *"/my-product/my-image1 "* ]]
+              [[ $(cat $(workspaces.data.path)/mock_curl.txt | head -n 2 | tail -n 1) \
+                  == *'"source_container_image_enabled":true}' ]]
+
+              [[ $(cat $(workspaces.data.path)/mock_curl.txt | head -n 3 | tail -n 1) \
+                  == *"/my-product/my-image2 "* ]]
+              [[ $(cat $(workspaces.data.path)/mock_curl.txt | head -n 4 | tail -n 1) \
+                  != *'"source_container_image_enabled":true}' ]]
```

Comment:
The second component has `"pushSourceContainer": "false"` so it should be called with `"source_container_image_enabled":false`, no? Or does the default override it? I wouldn't expect that, you should be able to have a default value and then override it on a component.

Reply:
It is just missing, not set with false AFAICT
`PAYLOAD=$(jq -c '. += {"source_container_image_enabled":true}' <<< $PAYLOAD)` 
I don't see any `PAYLOAD=$(jq -c '. += {"source_container_image_enabled":false}' <<< $PAYLOAD)` in the task code

Reply:
Oh, I missed the detail that here it's `!=` instead of `==`, my bad.